src/wire/igmp.rs

```diff
@@ -0,0 +1,262 @@
+/// Internet Group Management Protocol v2
+/// defined in [RFC_2236]
+///
+///    0                   1                   2                   3
+///    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+///   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+///   |      Type     | Max Resp Time |           Checksum            |
+///   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+///   |                         Group Address                         |
+///   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+///
+
+use core::{cmp, fmt};
+use byteorder::{ByteOrder, NetworkEndian};
+
+use {Error, Result};
+use phy::ChecksumCapabilities;
+use super::ip::checksum;
+use super::{Ipv4Packet, Ipv4Repr};
+
+use wire::Ipv4Address;
+
+enum_with_unknown! {
+    /// Internet protocol control message type.
+    pub doc enum Igmpv2MessageType(u8) {
```

Comment:
... and when the file is called `igmpv2` you can remove the obnoxious prefixes. See how the other protocols do it.

Reply:
Fixed to look like for ICMP packets